src/components/Comment/index.tsx

```diff
@@ -1,21 +1,61 @@
 import * as React from 'react';
+import { connect } from 'react-redux';
 import Textarea from 'react-textarea-autosize';
 import { Button, Form } from 'react-bootstrap';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 
-import { Comment } from '../../reducers/comments';
+import { ConnectedReduxProps } from '../../configureStore';
+import { ApplicationState } from '../../reducers';
+import {
+  Comment,
+  createCommentKey,
+  manageComment,
+} from '../../reducers/comments';
 import { gettext, sanitizeHTML, nl2br } from '../../utils';
 import styles from './styles.module.scss';
 
 type TextareaRef = React.RefObject<HTMLTextAreaElement> | undefined;
 
 export type PublicProps = {
-  comment: Comment | null;
-  createTextareaRef?: () => TextareaRef;
+  addonId: number;
+  commentId: number | null;
+  fileName: string | null;
+  line: number | null;
   readOnly: boolean;
+  versionId: number;
+};
+
+export type DefaultProps = {
+  _manageComment: typeof manageComment;
+  createTextareaRef?: () => TextareaRef;
 };
 
-export class CommentBase extends React.Component<PublicProps> {
+type PropsFromState = {
+  initialComment: Comment | null;
+  initialCommentText: string | null;
```

Comment:
TBH I find this whole `initialComment` and `initialCommentText` thing to be quite confusing. After reading all the code and tests I'm still not sure why we need `initialCommentText`. I was hoping the tests would make it clear, but they do not (to me anyway), which suggests that maybe there is some coverage missing. I did make a comment about a specific case not being covered, but I am concerned by how confused I am about this.

From what I understand, `initialCommentText` is about populating the form with data from `pendingCommentText` in the redux store, but that only seems to be relevant after a user has submitted the form. `initialCommentText` is only accessed in the `constructor`, so I assume it will only ever be used when the component is initially created, and won't ever be used on an update/re-render. So I don't really understand the flow here: when the form is submitted, which will cause `pendingCommentText` to be updated in the redux store, will that cause the `constructor` to rerun? My assumption is "no", but maybe I don't understand the lifecycle of the component. If I am correct that the `constructor` won't rerun, then I don't see how having access to `initialCommentText` in `props` is useful at that point. 

The tests show that you can do an initial render of the component with a value for `pendingCommentText` in the store, and it will use it, but this seems like it's not the way the component will actually behave. I assume that the component will be initially rendered with either an existing comment, or with no `pendingCommentText` in the store, and that only after the form is submitted will a value for `pendingCommentText` be added to the store.

Reply:
> when the form is submitted, which will cause `pendingCommentText` to be updated in the redux store, will that cause the `constructor` to rerun? My assumption is "no", but maybe I don't understand the lifecycle of the component. 

Yes, it will cause the constructor to run because this:

https://github.com/mozilla/addons-code-manager/blob/0b12496b0e2d78b6f792a5b801554e1e93de379e/src/reducers/comments.tsx#L162

...causes a re-render of `CodeView` and that creates a new instance of `Comment`. I'm not exactly sure *why* but it's probably [this render prop](https://github.com/mozilla/addons-code-manager/blob/0b12496b0e2d78b6f792a5b801554e1e93de379e/src/components/LinterProvider/index.tsx#L87).

I had to add this feature precisely because of this behavior, which is surprising. If an update to `pendingCommentText` were to instead update an existing component instance then we wouldn't need the feature at all. The edited comment would be held in state and the "pending" text would already be there. For this same reason, I did not implement a prop update of `initialComment` or `initialCommentText` because that should not be allowed (the internal state should take precedence).

This will probably become more clear when I land the next patch that integrates everything into `CodeView`. I can always do follow-ups later once the big picture comes into focus.